.circleci/config.yml

```diff
@@ -77,31 +51,37 @@ jobs:
     <<: *python_only_config
     steps:
       - checkout
-      - *save_repo
+      - persist_to_workspace:
+          root: "~/"
+          paths:
+            - marsha
 
   # install the project code and dependencies and save the venv and pip cache
   install_code:
     <<: *python_only_config
     steps:
-      - *restore_repo
+      - *attach_workspace
       - *restore_pip_cache
-      - *restore_venv
       - run:
           name: Install code
           command: |
             python -m venv ~/venv
             source ~/venv/bin/activate
             pip install --upgrade pip
             pip install -e .[dev]
-      - *save_venv
       - *save_pip_cache
+      - persist_to_workspace:
+          root: "~/"
+          paths:
+            - venv
+            - marsha
```

Comment:
I've never used workspaces, this is a brilliant idea! From what I've read [1], this will also speed up the build as it's faster than using the cache. :+1: 

[1] https://hackernoon.com/circleci-performance-difference-between-cache-and-workspace-5567679c3601

Reply:
Yes it's the point and it is exactly what I was looking for. Also, it is only update for next jobs if `persist_to_workspace` is used, it's why I do not persist it after lint jobs.